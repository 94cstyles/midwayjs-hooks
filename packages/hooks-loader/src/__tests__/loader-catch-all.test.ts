import { resolve } from 'path'
import compiler from './compiler'
import webpack from 'webpack'

const root = resolve(__dirname, './fixtures/catch-all')

const resolveEntry = (path: string) => {
  return resolve(__dirname, root, 'src/apis/', path)
}
const getOutput = (stats: webpack.Stats) => {
  return stats.toJson().modules[2].source
}

describe('hooks-loader', () => {
  test('Compile render', async () => {
    const stats = await compiler(resolveEntry('render/[...index].ts'), root)
    const output = getOutput(stats)
    expect(output).toMatchSnapshot()
  })

  test.skip('Compile lambda', async () => {
    const stats = await compiler(resolveEntry('lambda/index.ts'), root)
    const output = getOutput(stats)
    expect(output).toMatchSnapshot()
  })
})
