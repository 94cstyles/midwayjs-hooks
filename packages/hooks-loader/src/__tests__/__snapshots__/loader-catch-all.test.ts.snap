// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks-loader Compile lambda 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  default  function $default (...args) {
  return request({
    url: '/api',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"lambda-index\\",
  \\"gateway\\": \\"http\\",
  \\"functionGroup\\": \\"midway-faas-react-demo\\"
}
  })
}

"
`;

exports[`hooks-loader Compile render 1`] = `
"import { request } from '@midwayjs/hooks/request'


export  default  function $default (...args) {
  return request({
    url: '',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"render-index\\",
  \\"gateway\\": \\"http\\",
  \\"functionGroup\\": \\"midway-faas-react-demo\\"
}
  })
}

export  function foo (...args) {
  return request({
    url: '/foo',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"render-index-foo\\",
  \\"gateway\\": \\"http\\",
  \\"functionGroup\\": \\"midway-faas-react-demo\\"
}
  })
}

export  function bar (...args) {
  return request({
    url: '/bar',
    method: 'GET',
    data: {
      args: args,
    },
    meta: {
  \\"functionName\\": \\"render-index-bar\\",
  \\"gateway\\": \\"http\\",
  \\"functionGroup\\": \\"midway-faas-react-demo\\"
}
  })
}

"
`;
